metrics/src/poh_timing_point.rs

```diff
@@ -0,0 +1,205 @@
+        }
+    }
+}
+
+/// SlotPohTimingInfo. This struct is sent to channel and received by
+/// poh_timing_report service.
+#[derive(Clone, Debug)]
+pub struct SlotPohTimingInfo {
+    /// current slot
+    pub slot: Slot,
+    /// root slot
+    pub root_slot: Option<Slot>,
+    /// timing event
+    pub timing_point: PohTimingPoint,
+}
+
+impl fmt::Display for SlotPohTimingInfo {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "PohTimingPoint: {}, slot={}, root_slot={}",
+            self.timing_point,
+            self.slot,
+            self.root_slot.unwrap_or(0),
+        )
+    }
+}
+
+/// create slot poh timing point w/o root info.
+#[macro_export]
+macro_rules! create_slot_poh_time_point {
+    ($slot:expr, $timestamp:expr) => {
+        $crate::poh_timing_point::SlotPohTimingInfo {
+            slot: $slot,
+            root_slot: None,
+            timing_point: $timestamp,
+        }
+    };
+
+    ($slot:expr, $root_slot:expr, $timestamp:expr ) => {
+        $crate::poh_timing_point::SlotPohTimingInfo {
+            slot: $slot,
+            root_slot: Some($root_slot),
+            timing_point: $timestamp,
+        }
+    };
+}
+
+/// create slot poh start timing point w/o root info.
+#[macro_export]
+macro_rules! create_slot_poh_start_time_point {
+    ($slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $crate::poh_timing_point::PohTimingPoint::PohSlotStart($timestamp)
+        )
+    };
+    ($slot:expr, $root_slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $root_slot,
+            $crate::poh_timing_point::PohTimingPoint::PohSlotStart($timestamp)
+        )
+    };
+}
+
+/// create slot poh end timing point w/o root info.
+#[macro_export]
+macro_rules! create_slot_poh_end_time_point {
+    ($slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $crate::poh_timing_point::PohTimingPoint::PohSlotEnd($timestamp)
+        )
+    };
+    ($slot:expr, $root_slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $root_slot,
+            $crate::poh_timing_point::PohTimingPoint::PohSlotEnd($timestamp)
+        )
+    };
+}
+
+/// create slot poh full timing point w/o root info.
+#[macro_export]
+macro_rules! create_slot_poh_full_time_point {
+    ($slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $crate::poh_timing_point::PohTimingPoint::FullSlotReceived($timestamp)
+        )
+    };
+    ($slot:expr, $root_slot:expr, $timestamp:expr) => {
+        $crate::create_slot_poh_time_point!(
+            $slot,
+            $root_slot,
+            $crate::poh_timing_point::PohTimingPoint::FullSlotReceived($timestamp)
+        )
+    };
+}
```

Comment:
These seem like they could be constructors inside an `impl SlotPohTimingInfo` instead of macros

Reply:
yup. macros is an overkill for this. will update.